locals.tf

```diff
@@ -19,7 +19,14 @@ locals {
   public_subnet_ids  = var.public_subnet_ids
 
   enable_workers            = length(var.self_managed_node_groups) > 0 || length(var.managed_node_groups) > 0 ? true : false
-  worker_security_group_ids = local.enable_workers ? compact(flatten([[module.aws_eks.node_security_group_id], var.worker_additional_security_group_ids])) : []
+  worker_security_group_ids = local.enable_workers ? compact(flatten([[module.aws_eks.cluster_primary_security_group_id, module.aws_eks.node_security_group_id], var.worker_additional_security_group_ids])) : []
```

Comment:
this looks massively complex and brittle. Instead of forcing this upon users, I think educating users of the various options and "gotchas" to be aware of would be more beneficial